node/src/Pos/Delegation/Logic/VAR.hs

```diff
@@ -344,26 +335,24 @@ dlgVerifyBlocks ::
     => OldestFirst NE (Block ssc)
     -> m (Either Text (OldestFirst NE DlgUndo))
 dlgVerifyBlocks blocks = do
-    _dvCurEpoch <- GS.getThisEpochPostedKeys
-    let initState = DlgVerState _dvCurEpoch
     (richmen :: HashSet StakeholderId) <-
         lrcActionOnEpochReason
         headEpoch
         "Delegation.Logic#delegationVerifyBlocks: there are no richmen for current epoch"
         LrcDB.getRichmenDlg
-    flip evalStateT initState . evalMapCede mempty . runExceptT $
+    evalMapCede mempty . runExceptT $
         mapM (verifyBlock richmen) blocks
   where
     headEpoch = blocks ^. _Wrapped . _neHead . epochIndexL
 
     verifyBlock ::
         HashSet StakeholderId ->
         Block ssc ->
-        ExceptT Text (MapCede (StateT DlgVerState m)) DlgUndo
+        ExceptT Text (MapCede m) DlgUndo
     verifyBlock _ (Left genesisBlk) = do
-        prevThisEpochPosted <- use dvCurEpoch
-        dvCurEpoch .= HS.empty
         let blkEpoch = genesisBlk ^. epochIndexL
+        prevThisEpochPosted <- getAllPostedThisEpoch
+        mapM_ delThisEpochPosted prevThisEpochPosted
```

Comment:
can you explain this code? do you want to set to an empty set of `posted this epoch certs`? why?